src/appkit.rs

```diff
@@ -381,6 +392,285 @@ impl NSRunningApplication for id {
+    }
+
+    unsafe fn setData_forType(self, data: id, _type: id) -> BOOL {
+        msg_send![self, setData:data forType:_type]
+    }
+
+    unsafe fn setString_forType(self, string: id, _type: id) -> BOOL {
+        msg_send![self, setString:string forType:_type]
+    }
+
+    unsafe fn setPropertyList_forType(self, propertyList: id, _type: id) -> BOOL {
+        msg_send![self, setPropertyList:propertyList forType:_type]
+    }
+
+    unsafe fn dataForType(self, _type: id) -> id {
+        msg_send![self, dataForType:_type]
+    }
+
+    unsafe fn stringForType(self, _type: id) -> id {
+        msg_send![self, stringForType:_type]
+    }
+
+    unsafe fn propertyListForType(self, _type: id) -> id {
+        msg_send![self, propertyListForType:_type]
+    }
+}
+
+pub trait NSPasteboardItemDataProvider {
+    unsafe fn pasteboard_item_provideDataForType(self, pasteboard: id, item: id, _type: id);
+    unsafe fn pasteboardFinishedWithDataProvider(self, pasteboard: id);
+}
+
+impl NSPasteboardItemDataProvider for id {
+    unsafe fn pasteboard_item_provideDataForType(self, pasteboard: id, item: id, _type: id) {
+        msg_send![self, pasteboard:pasteboard item:item provideDataForType:_type]
+    }
+
+    unsafe fn pasteboardFinishedWithDataProvider(self, pasteboard: id) {
+        msg_send![self, pasteboardFinishedWithDataProvider:pasteboard]
+    }
+}
+
+pub trait NSPasteboardWriting {
+    unsafe fn writableTypesForPasteboard(self, pasteboard: id) -> id;
+    unsafe fn writingOptionsForType_pasteboard(self, _type: id, pasteboard: id) -> NSPasteboardWritingOptions;
+
+    unsafe fn pasteboardPropertyListForType(self, _type: id) -> id;
+}
+
+impl NSPasteboardWriting for id {
+    unsafe fn writableTypesForPasteboard(self, pasteboard: id) -> id {
+        msg_send![self, writableTypesForPasteboard:pasteboard]
+    }
+
+    unsafe fn writingOptionsForType_pasteboard(self, _type: id, pasteboard: id) -> NSPasteboardWritingOptions {
+        msg_send![self, writingOptionsForType:_type pasteboard:pasteboard]
+    }
+
+    unsafe fn pasteboardPropertyListForType(self, _type: id) -> id {
+        msg_send![self, pasteboardPropertyListForType:_type]
+    }
+}
+
+pub trait NSPasteboardReading {
+    unsafe fn initWithPasteboardPropertyList_ofType(self, propertyList: id, _type: id) -> id;
+
+    unsafe fn readableTypesForPasteboard(self, pasteboard: id) -> id;
+    unsafe fn readingOptionsForType(self, pasteboard: id);
+}
+
+impl NSPasteboardReading for id {
+    unsafe fn initWithPasteboardPropertyList_ofType(self, propertyList: id, _type: id) -> id {
+        msg_send![self, initWithPasteboardPropertyList:propertyList ofType:_type]
+    }
+
+    unsafe fn readableTypesForPasteboard(self, pasteboard: id) -> id {
+        let class: id = msg_send![self, class];
+        msg_send![class, readableTypesForPasteboard:pasteboard]
+    }
+    unsafe fn readingOptionsForType(self, pasteboard: id)  {
```

Comment:
https://developer.apple.com/reference/appkit/nspasteboardreading/1533559-readingoptionsfortype?language=objc says this should return a value.